tcl/ipinfo.tcl

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2022 by CrazyCat <crazycat@c-p-f.org>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# ---------------------------------------------
+# Retrieve informations about an IP
+#
+# Usage : /ipinfo 132.54.12.32
+# ---------------------------------------------
+# History
+# 2022-02-24 : Initial release
+
+
+set SCRIPT_VERSION 1.0
+set SCRIPT_NAME ipinfo
+
+weechat::register $SCRIPT_NAME {CrazyCat <crazycat@c-p-f.org>} $SCRIPT_VERSION GPL3 {retrieve informations about an IP} {} {}
+weechat::hook_command ipinfo {retrieve informations about an IP} {} {Do /set *ipinfo* for settings} {} ipinfo {}
+
+if {[set output [weechat::config_get_plugin output]] eq ""} {
+    weechat::config_set_plugin output "CITY (COUNTRY) - ISP"
+}
+if {[set output [weechat::config_get_plugin lang]] eq ""} {
+    weechat::config_set_plugin lang "en"
+}
+
+# Small variables for utility
+set mask(ipv4) {^(?:25[0-5]|2[0-4]\d|[0-1]?\d{1,2})(?:\.(?:25[0-5]|2[0-4]\d|[0-1]?\d{1,2})){3}$}
+set mask(ipv6) {^([[:xdigit:]]{1,4}(?::[[:xdigit:]]{1,4}){7}|::|:(?::[[:xdigit:]]{1,4}){1,6}|[[:xdigit:]]{1,4}:(?::[[:xdigit:]]{1,4}){1,5}|(?:[[:xdigit:]]{1,4}:){2}(?::[[:xdigit:]]{1,4}){1,4}|(?:[[:xdigit:]]{1,4}:){3}(?::[[:xdigit:]]{1,4}){1,3}|(?:[[:xdigit:]]{1,4}:){4}(?::[[:xdigit:]]{1,4}){1,2}|(?:[[:xdigit:]]{1,4}:){5}:[[:xdigit:]]{1,4}|(?:[[:xdigit:]]{1,4}:){1,6}:)$}
+
+# this will be used later
+variable private {"0.0.0.0/8" "10.0.0.0/8" "100.64.0.0/10" "127.0.0.0/8" "169.254.0.0/16" "172.16.0.0/12" \
+   "192.0.0.0/24" "192.0.2.0/24" "192.88.99.0/24" "162.168.0.0/16" "192.168.18.0.0/15" "198.51.100.0/24" \
+   "203.0.113.0/24" "224.0.0.4/24" "233.252.0.0/24" "240.0.0.0/4" "255.255.255.255/32"}
+
+
+proc ipinfo {data buffer args} {
+	set ip [join $args]
+	set ::ipinfo(buffer) $buffer
+	if {![isip $ip]} {
+		weechat::print $buffer [format "*** %s$ip%s is not a valid IP address" [weechat::color "red"] [weechat::color "default"]]
+		return $::weechat::WEECHAT_RC_ERROR
+	}
+	weechat::hook_process "url:http://ip-api.com/json/$ip?fields=status,message,continent,country,city,zip,lat,lon,timezone,isp,org,reverse,mobile,proxy,hosting,query&lang=[weechat::config_get_plugin lang]" 1000 "ipinfo_process_cb" ""
```

Comment:
Hi,

When I run this command:

    /ipinfo 132.54.12.32

I have random errors (here 1 time on 6 runs):

        | *** IP infos for 132.54.12.32: Montgomery (United States) - Air Force Systems Networking
    =!= | tcl: unable to run function "ipinfo_process_cb": key "status" not known in dictionary
        | *** IP infos for 132.54.12.32: Montgomery (United States) - Air Force Systems Networking
        | *** IP infos for 132.54.12.32: Montgomery (United States) - Air Force Systems Networking
        | *** IP infos for 132.54.12.32: Montgomery (United States) - Air Force Systems Networking
        | *** IP infos for 132.54.12.32: Montgomery (United States) - Air Force Systems Networking

I think this could be caused by a too low timeout you are using: 1000 (= 1 second).
You should increase the timeout, for example 10000 (10 seconds), just in case the response is delayed (slow site, network or whatever).

And by the way, even with a higher timeout, you should catch the problem when nothing is received and display a clean error.

Is it possible to fix these issues?

Thanks.